contracts/OXO.sol

```diff
@@ -0,0 +1,202 @@
+        } else {
+            payReward(gameId);
+            GameFinished(gameId);
+        }
+
+        return true;
+    }
+
+    function gameStatus(uint gameId) constant public returns(GameStatus status, address currentPlayer, address winnerPlayer, uint8[] player1moves, uint8[] player2moves) {
+        return (
+            games[gameId].status, 
+            games[gameId].currentPlayer, 
+            games[gameId].winnerPlayer, 
+            games[gameId].moves[games[gameId].player1],
+            games[gameId].moves[games[gameId].player2]
+        );
+    }
+
+    function cancelGame(uint gameId) public returns(bool) {
+        require(games[gameId].player1 == msg.sender || games[gameId].player2 == msg.sender);
+        require(games[gameId].currentPlayer != msg.sender);
+        require(games[gameId].status != GameStatus.Finished);
+        require(games[gameId].currentTime + gameTimeOut > block.timestamp);
+
+        games[gameId].status = GameStatus.Finished;
+        games[gameId].winnerPlayer = msg.sender;
+        payReward(gameId);
+
+        GameCanceled(gameId);
+        
+        return true;
+    } 
+
+    function isMoveUnique(uint gameId, uint fieldNumber) constant private returns(bool) {
+        Game storage game = games[gameId];
+        // check player1 moves
+        for (uint i = 0; i < game.moves[game.player1].length; i++) {
+            if (game.moves[game.player1][i] == fieldNumber)
+                return false;
+        }
+        // check player2 moves
+        for (uint j = 0; j < game.moves[game.player2].length; j++) {
+            if (game.moves[game.player2][j] == fieldNumber)
+                return false;
+        }
+        return true;
+    }
+
+    function checkGame(uint gameId) private {
+        Game storage game = games[gameId];
+        // no sence calculate game if players not have 3 moves
+        if (game.moves[game.player1].length < 3) 
+            return;
+
+        // check if one of winConbinations has players moves
+        for (uint a = 0; a < winCombinations.length; a++) {
+            uint combCount = 0;
+            for (uint b = 0; b < 3; b++) {
+                for (uint i = 0; i < game.moves[game.currentPlayer].length; i++) {
+                    if (game.moves[game.currentPlayer][i] == winCombinations[a][b]) {
+                        combCount++;
+                        break;
+                    }
+                }
+            }
+            if (combCount == 3) {
+                game.winnerPlayer = game.currentPlayer;
+                game.status = GameStatus.Finished;
+                Message("Player won!");
+                break;
+            }
+        }
+        // we don't have winner if it is a fifth move
+        if (game.moves[game.currentPlayer].length == 5)
+            game.status = GameStatus.Finished;
+    }
+
+    function payReward(uint gameId) constant private {
+        Game storage game = games[gameId];
+        require(game.status == GameStatus.Finished);
```

Comment:
There is no situation when payReward() is called before game has finished. You can safely remove this excessive check and save some gas.

Reply:
lets save this gas)